fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/TransactionImpl.java

```diff
@@ -97,7 +98,9 @@ public synchronized void commit() {
             }
 
             updateState(TransactionState.COMMITTED);
-            this.getEventAccumulator().emitEvents(this, baseUri, userAgent);
+            if (!this.suppressEvents) {
+                this.getEventAccumulator().emitEvents(this, baseUri, userAgent);
```

Comment:
This may cause a memory leak due to the events never being cleared from the accumulator.